client/gutenberg/extensions/giphy/edit.js

```diff
@@ -0,0 +1,193 @@
+		);
+		if ( match ) {
+			giphyID = match[ 1 ];
+		}
+		if ( giphyID ) {
+			return this.fetch( this.urlForId( giphyID ) );
+		}
+
+		return this.fetch( this.urlForSearch( searchText ) );
+	}, 250 );
+
+	urlForSearch = searchText => {
+		const { apiKey } = settings;
+		return `//api.giphy.com/v1/gifs/search?q=${ encodeURIComponent(
+			searchText
+		) }&api_key=${ apiKey }&limit=1`;
+	};
+
+	urlForId = giphyId => {
+		const { apiKey } = settings;
+		return `//api.giphy.com/v1/gifs/${ giphyId }?api_key=${ apiKey }`;
+	};
+
+	splitAndLast = ( array, delimiter ) => {
+		const split = array.split( delimiter );
+		return split[ split.length - 1 ];
+	};
+
+	fetch = url => {
+		const { setAttributes } = this.props;
+		const xhr = new XMLHttpRequest();
+		xhr.open( 'GET', url );
+		xhr.onload = () => {
+			if ( xhr.status === 200 ) {
+				const res = JSON.parse( xhr.responseText );
+				const giphyData = res.data.length > 0 ? res.data[ 0 ] : res.data;
+				// No results
+				if ( ! giphyData.images ) {
+					return;
+				}
+				const topPadding = Math.floor(
+					( giphyData.images.original.height / giphyData.images.original.width ) * 100
+				);
+				const giphyUrl = giphyData.embed_url;
+				setAttributes( { giphyUrl, topPadding } );
+				this.maintainFocus( 500 );
+			} else {
+				// Error handling TK
+			}
+		};
+		xhr.send();
+	};
+
+	setFocus = () => {
+		this.maintainFocus();
+		this.textControlRef.current.querySelector( 'input' ).focus();
+	};
+
+	maintainFocus = ( timeoutDuration = 3500 ) => {
+		this.setState( { focus: true }, () => {
+			if ( this.timer ) {
+				clearTimeout( this.timer );
+			}
+			if ( this.hasSearchText() ) {
+				this.timer = setTimeout( () => {
+					this.setState( { focus: false } );
+				}, timeoutDuration );
+			}
+		} );
+	};
+
+	clearFocus = () => {
+		this.setState( { focus: false }, () => {
+			if ( this.timer ) {
+				clearTimeout( this.timer );
+			}
+		} );
+	};
+
+	hasSearchText = () => {
+		const { attributes } = this.props;
+		const { searchText } = attributes;
+		return searchText && searchText.length > 0;
+	};
+
+	render() {
+		const { attributes, className, isSelected, setAttributes } = this.props;
+		const { align, caption, giphyUrl, searchText, topPadding } = attributes;
+		const { focus } = this.state;
+		const style = {
+			paddingTop: `${ topPadding }%`,
+		};
+		const classes = classNames( className, `align${ align }` );
+		const textControlClasses = classNames(
+			'wp-block-jetpack-giphy_text-input-field',
+			focus || ! this.hasSearchText() ? 'has-focus' : 'no-focus'
+		);
+		return (
+			<Fragment>
+				<BlockControls>
+					<BlockAlignmentToolbar
+						onChange={ value => setAttributes( { align: value } ) }
+						value={ align }
+					/>
+				</BlockControls>
+				<div className={ classes }>
+					<figure style={ style }>
+						<div
+							className="wp-block-jetpack-giphy_cover"
+							onClick={ this.setFocus }
+							onKeyDown={ this.setFocus }
```

Comment:
Can you explain why this is necessary?

Reply:
In order to present the selected gif unobscured, the text input disappears after 3.5s of inactivity. Clicking on the block makes it visible again, and sets keyboard focus to the input, which explains the `onClick` handler. The `onKeyDown` handler is present for this similar scenario:

- User starts typing, pauses for 4 seconds to consider exact what animated gif they want to show the world
- Because of timeout, the input disappears
- When they start typing again, input reappears, and the field regains focus.

Reply:
I overstepped in ae6ae7cd2275b1fa880830d658562ae06bcbb40c by making the focused input always visible.

I'll roll that back and this makes sense again 👍

Reply:
Rolled https://github.com/Automattic/wp-calypso/commit/ae6ae7cd2275b1fa880830d658562ae06bcbb40c back in https://github.com/Automattic/wp-calypso/pull/29638/commits/24f732045499a6f596b8e24cbaa1de5ccbe99249. All fixed!